pkg/tool/self/self.go

```diff
@@ -0,0 +1,157 @@
+		return fmt.Errorf("failed to find valid backplane-tools binary asset")
+	}
+	if checksumAsset.GetName() == "" {
+		return fmt.Errorf("failed to find valid backplane-tools checksum asset")
+	}
+
+	// Download the arch- & os-specific assets
+	toolDir := t.toolDir(rootDir)
+	versionedDir := filepath.Join(toolDir, release.GetTagName())
+	err = os.MkdirAll(versionedDir, os.FileMode(0755))
+	if err != nil {
+		return fmt.Errorf("failed to create version-specific directory '%s': %w", versionedDir, err)
+	}
+
+	err = t.source.DownloadReleaseAssets([]*gogithub.ReleaseAsset{checksumAsset, backplaneArchiveAsset}, versionedDir)
+	if err != nil {
+		return fmt.Errorf("failed to download one or more assets: %w", err)
+	}
+
+	// Verify checksum of downloaded assets
+	backplaneArchiveFilepath := filepath.Join(versionedDir, backplaneArchiveAsset.GetName())
+	binarySum, err := utils.Sha256sum(backplaneArchiveFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to calculate checksum for '%s': %w", backplaneArchiveFilepath, err)
+	}
+
+	checksumFilePath := filepath.Join(versionedDir, checksumAsset.GetName())
+	checksumLine, err := utils.GetLineInFile(checksumFilePath, backplaneArchiveAsset.GetName())
+	if err != nil {
+		return fmt.Errorf("failed to retrieve checksum from file '%s': %w", checksumFilePath, err)
+	}
+	checksumTokens := strings.Fields(checksumLine)
+	if len(checksumTokens) != 2 {
+			return fmt.Errorf("the checksum file '%s' is invalid: expected 2 fields, got %d", checksumFilePath, len(checksumTokens))
+	}
+	actual := checksumTokens[0]
+
+	if strings.TrimSpace(binarySum) != strings.TrimSpace(actual) {
+		return fmt.Errorf("WARNING: Checksum for backplane-tools does not match the calculated value. Please retry installation. If issue persists, this tool can be downloaded manually at %s\n", backplaneArchiveAsset.GetBrowserDownloadURL())
+	}
+
+	// Untar binary bundle
+	err = utils.Unarchive(backplaneArchiveFilepath, versionedDir)
+	if err != nil {
+		return fmt.Errorf("failed to unarchive the backplane-tools asset file '%s': %w", backplaneArchiveFilepath, err)
+	}
+
+	// Link as latest
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove existing 'backplane-tools' binary at '%s': %w", latestDir, err)
+	}
+
+	backplaneBinaryFilepath := filepath.Join(versionedDir, "backplane-tools")
+	err = os.Symlink(backplaneBinaryFilepath, latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to link new 'backplane-tools' binary to '%s': %w", latestDir, err)
+	}
+	return nil
+}
+
+// toolDir returns this tool's specific directory given the root directory all tools are installed in
+func (t *Tool) toolDir(rootDir string) string {
+	return filepath.Join(rootDir, "backplane-tools")
+}
+
+// symlinkPath returns the path to the symlink created by this tool, given the latest directory
+func (t *Tool) symlinkPath(latestDir string) string {
```

Comment:
I'll look into addressing this [here](https://github.com/openshift/backplane-tools/pull/16#discussion_r1348907578). For now, since this is how the other tools define their directories, lets keep it the same.